lib/client.js

```diff
@@ -78,6 +81,11 @@ module.exports = function (dependencies) {
               this.config.token.regenerate(tokenGeneration);
               resolve(this.write(notification, device, retryCount + 1));
               return;
+            } else if (status === "500" && response.reason === "InternalServerError") {
+              stream.connection.close();
+              let error = new VError("Error 500, stream ended unexpectedly");
```

Comment:
If this is the case, I think a common return pattern MUST be defined, otherwise, there would be three different types of return pattern including @jpwilliams given above.
`{ device, status, response, error }, 
{ device, status, response}, 
{ device, error }`
It would make user confused.